app/datatables/protocols_datatable.rb

```diff
@@ -77,88 +80,84 @@ def new_search_condition(column, value)
 
   # Returns json of current protocols (already paginated)
   def data
-    result_data = []
-    records.each do |record|
-      protocol = Protocol.find(record.id)
-      result_data << {
-        'DT_RowId': record.id,
-        'DT_RowAttr': {
-          'data-permissions-url': permissions_protocol_path(protocol)
-        },
-        '1': if protocol.in_repository_archived?
-               escape_input(record.name)
-             else
-               name_html(record)
-             end,
-        '2': keywords_html(record),
-        '3': modules_html(record),
-        '4': escape_input(record.full_username_str),
-        '5': timestamp_column_html(record),
-        '6': I18n.l(record.updated_at, format: :full)
+    records.map do |record|
+      {
+        DT_RowId: record.id,
+        DT_CanClone: can_clone_protocol_in_repository?(record),
+        DT_CloneUrl: if can_clone_protocol_in_repository?(record)
+                       clone_protocol_path(record, team: @team, type: @type)
+                     end,
+        DT_CanMakePrivate: can_manage_protocol_in_repository?(record),
+        DT_CanPublish: can_manage_protocol_in_repository?(record),
+        DT_CanArchive: can_manage_protocol_in_repository?(record),
+        DT_CanRestore: can_restore_protocol_in_repository?(record),
+        DT_CanExport: can_read_protocol_in_repository?(record),
```

Comment:
I think we don't need it anymore, I made permissions the same as in other places. When we load permissions on row selection.

Reply:
OK